CK3toEU4/Source/EU4World/Religion/GeneratedReligion.h

```diff
@@ -6,24 +6,26 @@
 
 namespace EU4
 {
-class GeneratedReligion
-{
-  public:
-	std::string country;
-	std::string countrySecondary;
-	std::string province;
-	std::string unique;
-	std::string nonUnique;
-	std::string iconPath;
-	std::string religionGroup;
-	std::string staticBlob;
-	std::string name;
-	std::string allowedConversion;
-	std::string parent;
-	std::optional<commonItems::Color> color;
-	int icon = 0;
-	friend std::ostream& operator<<(std::ostream& output, const GeneratedReligion& religion);
-};
+	class GeneratedReligion
+	{
+	  public:
+		void EU4::GeneratedReligion::returnRebels(std::ostream& output) const;
+
+		std::string country;
+		std::string countrySecondary;
+		std::string province;
+		std::string unique;
+		std::string nonUnique;
+		std::string iconPath;
+		std::string religionGroup;
+		std::string staticBlob;
+		std::string name;
+		std::string allowedConversion;
+		std::string parent;
+		std::optional<commonItems::Color> color;
+		int icon = 0;
+		friend std::ostream& operator<<(std::ostream& output, const GeneratedReligion& religion);
```

Comment:
I concur, this is just raw storage, can be a struct.

Reply:
Showing my ignorance again. A what?

Reply:
struct Generated religion {
int this
string that
}

It's a container similar to class but without private/public separation. It can have methods same as class. We use it when we need to pack some data in a convenient container.

Reply:
Why use it instead of a class? Does it take up less memory or some-such?

Reply:
Because when constructing this I was overzealous in promoting it to a class, where in fact all I needed was a simple container. It's simpler, doesn't require a friend function, it has no methods and yes, takes less memory. just change class to struct and fix if something breaks.

Reply:
Beyond that, it better indicates the intent of the code: https://www.fluentcpp.com/2017/06/13/the-real-difference-between-struct-class/